deepset_cloud_sdk/_utils/ratelimiter.py

```diff
@@ -0,0 +1,58 @@
+import asyncio
+import time
+from typing import Any
+
+import aiohttp
+from aiohttp import ClientSession
+
+
+class RateLimiter:
+    """Rate limits an HTTP client for post() calls.
+
+    Calls are rate-limited by host.
+    This is a slightly modified version of the following code:
+      - https://gist.github.com/pquentin/28628ad59145e339027cc1612195a961
+    which is referenced in the following blog post:
+      - https://quentin.pradet.me/blog/how-do-you-rate-limit-calls-with-aiohttp.html
+    This class is not thread-safe.
+    """
+
+    RATE = 500  # 500 requests per second, AWS limit is 3500 PUT/POST/DELETE requests a second
+    MAX_TOKENS = 500
+
+    def __init__(self, client: ClientSession):
+        """
+        Create RateLimiter Object.
+
+        :param client: Instance of an aiohttp.ClientSession.
+        """
+        self.client = client
+        self.tokens = RateLimiter.MAX_TOKENS
+        self.updated_at = time.monotonic()
```

Comment:
can we make these protected attributes please unless they are publicly accessed?